app/components/job-tabs.js

```diff
@@ -1,17 +1,36 @@
 import Component from '@ember/component';
-import { match, not, reads } from '@ember/object/computed';
+import { computed } from '@ember/object';
+import { match, not, or, reads } from '@ember/object/computed';
 import { inject as service } from '@ember/service';
 import WithConfigValidation from 'travis/mixins/components/with-config-validation';
 
 export default Component.extend(WithConfigValidation, {
-  tagName: 'div',
   classNames: ['travistab'],
 
   router: service(),
+  modelName: reads('model.modelName'),
+
+  build: null,
+  job: null,
+  model: or('job', 'build'),
+  messages: reads('request.messages'),
 
   isConfig: match('router.currentRouteName', /config$/),
   isLog: not('isConfig'),
 
-  messages: reads('job.build.request.messages')
+  request: computed('job', 'build', function () {
+    if (this.build)
+      return this.build.get('request');
+    else
+      return this.job.get('build.request');
+  }),
```

Comment:
It looks like the following should work too:

```suggestion
  request: or('build.request', 'job.build.request'),
```

Reply:
woha, of course! 😆 

done: https://github.com/travis-ci/travis-web/commit/f20c4094c4311363723d2c94003c19f6ca636281